cmd/lotus-miner/sealing.go

```diff
@@ -101,17 +101,27 @@ var sealingWorkersCmd = &cli.Command{
 
 			ramBarsRes := int(stat.Info.Resources.MemReserved * barCols / stat.Info.Resources.MemPhysical)
 			ramBarsUsed := int(stat.MemUsedMin * barCols / stat.Info.Resources.MemPhysical)
-			ramBar := color.YellowString(strings.Repeat("|", ramBarsRes)) +
-				color.GreenString(strings.Repeat("|", ramBarsUsed)) +
-				strings.Repeat(" ", int(barCols)-ramBarsUsed-ramBarsRes)
+			ramRepeatSpace := int(barCols) - (ramBarsUsed + ramBarsRes)
+			var ramBar string
+			if ramRepeatSpace < 0 {
+				ramRepeatSpace = 0
+				ramBar = color.RedString(strings.Repeat("|", ramBarsRes)) +
+					color.GreenString(strings.Repeat("|", ramBarsUsed)) +
+					strings.Repeat(" ", ramRepeatSpace)
+			} else {
+				ramBar = color.YellowString(strings.Repeat("|", ramBarsRes)) +
+					color.GreenString(strings.Repeat("|", ramBarsUsed)) +
+					strings.Repeat(" ", ramRepeatSpace)
+			}
 
 			vmem := stat.Info.Resources.MemPhysical + stat.Info.Resources.MemSwap
 
 			vmemBarsRes := int(stat.Info.Resources.MemReserved * barCols / vmem)
 			vmemBarsUsed := int(stat.MemUsedMax * barCols / vmem)
+			vmemRepeatSpace := int(barCols) - (vmemBarsUsed + vmemBarsRes)
 			vmemBar := color.YellowString(strings.Repeat("|", vmemBarsRes)) +
 				color.GreenString(strings.Repeat("|", vmemBarsUsed)) +
-				strings.Repeat(" ", int(barCols)-vmemBarsUsed-vmemBarsRes)
+				strings.Repeat(" ", vmemRepeatSpace)
```

Comment:
Is this guaranteed to be non-negative?

Reply:
I am not 100% sure, so better be safe and keep the same pattern as `ramBar` for simplicity.